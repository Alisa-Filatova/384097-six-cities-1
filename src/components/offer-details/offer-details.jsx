import React from 'react';
import PropTypes from 'prop-types';
import ReviewsList from '../reviews-list/reviews-list.jsx';
import Map from '../map/map.jsx';
import OffersList from '../offers-list/offers-list.jsx';

const OfferDetails = ({offer, reviews, nearOffers}) => {
  return (
    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {offer.images.map((image, idx) =>
              <div className="property__image-wrapper" key={image + idx}>
                <img className="property__image" src={image} alt="Photo studio" />
              </div>
            )}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            {offer.is_premium &&
              <div className="property__mark">
                <span>Premium</span>
              </div>
            }
            <div className="property__name-wrapper">
              <h1 className="property__name">{offer.title}</h1>
              <button
                className={
                  `place-card__bookmark-button button 
                  ${offer.is_favorite ? `place-card__bookmark-button--active` : ``}`
                }
                type="button"
              >
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"/>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: (offer.rating * 10) * 2 + `%`}} />
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{offer.rating}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {offer.type}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {offer.bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                Max {offer.max_adults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">&euro;{offer.price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">
                {offer.goods.map((good) =>
                  <li key={good} className="property__inside-item">{good}</li>
                )}
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img
                    className="property__avatar user__avatar"
                    src={offer.host.avatar_url}
                    width="74"
                    height="74"
                    alt="Host avatar"
                  />
                </div>
                <span className="property__user-name">{offer.host.name}</span>
                {offer.host.is_pro && <span className="property__user-status">Pro</span>}
              </div>
              <div className="property__description">
                <p className="property__text">{offer.description}</p>
              </div>
            </div>
            <section className="property__reviews reviews">
              <ReviewsList reviews={reviews} />
              <form className="reviews__form form" action="#" method="post">
                <label className="reviews__label form__label" htmlFor="review">Your review</label>
                <div className="reviews__rating-form form__rating">
                  <input className="form__rating-input visually-hidden" name="rating" value="5" id="5-stars" type="radio"/>
                  <label htmlFor="5-stars" className="reviews__rating-label form__rating-label" title="perfect">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"/>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="4" id="4-stars" type="radio"/>
                  <label htmlFor="4-stars" className="reviews__rating-label form__rating-label" title="good">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"/>
                    </svg>
                  </label>

                  <input className="form__rating-input visually-hidden" name="rating" value="3" id="3-stars" type="radio"/>
                  <label htmlFor="3-stars" className="reviews__rating-label form__rating-label" title="not bad">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"/>
                    </svg>
                  </label>

                  <input
                    className="form__rating-input visually-hidden"
                    name="rating"
                    value="2"
                    id="2-stars"
                    type="radio"
                  />
                  <label htmlFor="2-stars" className="reviews__rating-label form__rating-label" title="badly">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"/>
                    </svg>
                  </label>

                  <input
                    className="form__rating-input visually-hidden"
                    name="rating"
                    value="1"
                    id="1-star"
                    type="radio"
                  />
                  <label htmlFor="1-star" className="reviews__rating-label form__rating-label" title="terribly">
                    <svg className="form__star-image" width="37" height="33">
                      <use xlinkHref="#icon-star"/>
                    </svg>
                  </label>
                </div>
                <textarea
                  className="reviews__textarea form__textarea"
                  id="review"
                  name="review"
                  placeholder="Tell how was your stay, what you like and what can be improved"
                />
                <div className="reviews__button-wrapper">
                  <p className="reviews__help">
                    To submit review please make sure to set <span className="reviews__star">rating</span> and describe your
                    stay with at least <b className="reviews__text-amount">50 characters</b>.
                  </p>
                  <button className="reviews__submit form__submit button" type="submit" disabled="">Submit</button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <Map
          className="property__map map"
          key={offer.city.name}
          currentCity={offer.city}
          activeOfferId={offer.id}
          cityOffers={nearOffers.slice(0, 3)}
        />
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <OffersList rentalOffers={nearOffers.slice(0, 3)} />
        </section>
      </div>
    </main>
  );
};

OfferDetails.propTypes = {
  offer: PropTypes.object.isRequired,
  reviews: PropTypes.arrayOf(PropTypes.object),
  nearOffers: PropTypes.arrayOf(PropTypes.object),
};

export default OfferDetails;
