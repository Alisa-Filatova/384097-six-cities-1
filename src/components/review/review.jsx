import React from 'react';
import PropTypes from 'prop-types';
import RatingStars from '../rating-stars/rating-stars.jsx';

const Review = ({review}) => {
  return (
    <li className="reviews__item">
      <div className="reviews__user user">
        <div className="reviews__avatar-wrapper user__avatar-wrapper">
          <img
            className="reviews__avatar user__avatar"
            src={review.user.avatarUrl}
            width="54"
            height="54"
            alt="Reviews avatar"
          />
        </div>
        <span className="reviews__user-name">{review.user.name}</span>
      </div>
      <div className="reviews__info">
        <RatingStars
          rating={review.rating}
          prefix="reviews"
        />
        <p className="reviews__text">{review.comment}</p>
        <time className="reviews__time" dateTime={review.date}>
          {new Date(review.date).toLocaleDateString(`en-US`, {month: `long`, year: `numeric`})}
        </time>
      </div>
    </li>
  );
};

Review.propTypes = {
  review: PropTypes.object.isRequired,
};

export default Review;
