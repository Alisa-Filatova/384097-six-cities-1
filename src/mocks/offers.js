export default [
  {
    id: 123,
    title: `Beautiful & luxurious apartment at great location`,
    img: `img/apartment-01.jpg`,
    isPremium: true,
    price: 120,
    stars: 4,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 234,
    title: `Wood and stone place`,
    img: `img/apartment-02.jpg`,
    isPremium: false,
    price: 80,
    stars: 2,
    type: `Private room`,
    isInBookmarks: true,
  },
  {
    id: 456,
    title: `Canal View Prinsengracht`,
    img: `img/apartment-03.jpg`,
    isPremium: false,
    price: 180,
    stars: 3,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 567,
    title: `Nice, cozy, warm big bed apartment`,
    img: `img/apartment-01.jpg`,
    isPremium: false,
    price: 100,
    stars: 4,
    type: `Private room`,
    isInBookmarks: true,
  },
];
