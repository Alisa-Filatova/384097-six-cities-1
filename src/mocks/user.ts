export const userMock = {
  avatarUrl: `/static/avatar/4.jpg`,
  email: `alicefill@y888.ru`,
  id: 1,
  isPro: false,
  name: `alicefill`,
};
