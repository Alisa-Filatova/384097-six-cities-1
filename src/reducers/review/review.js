import {ResponseStatus} from '../../enums/response-status';

const initialState = {
  reviewsList: [],
  postReviewStatus: null,
};

const ActionsType = {
  GET_REVIEWS: `GET_REVIEWS`,
  POST_REVIEW: `POST_REVIEW`,
  GET_POST_REVIEW_STATUS: `GET_POST_REVIEW_STATUS`,
};

const ActionCreator = {
  getReviews: (reviews) => ({
    type: ActionsType.GET_REVIEWS,
    payload: reviews,
  }),

  postReview: (reviews) => ({
    type: ActionsType.POST_REVIEW,
    payload: reviews,
  }),

  getPostReviewStatus: (postReviewStatus) => ({
    type: ActionsType.GET_POST_REVIEW_STATUS,
    payload: postReviewStatus,
  }),
};

const Operation = {
  getReviewsList: (id) => (dispatch, getState, api) => {
    return api.get(`/comments/${id}`)
      .then((response) => {
        dispatch(ActionCreator.getReviews(response.data));
      })
      .catch(() => {});
  },
  postReview: (id, review) => (dispatch, getState, api) => {
    return api.post(`/comments/${id}`, review)
      .then((response) => {
        dispatch(ActionCreator.postReview(response.data));
        dispatch(ActionCreator.getPostReviewStatus(response.status));
      })
      .catch((error) => {
        if (error.response.status === ResponseStatus.BAD_REQUEST) {
          dispatch(ActionCreator.getPostReviewStatus(error.response.status));
        } else if (error.response.status === ResponseStatus.FORBIDDEN) {
          dispatch(ActionCreator.getPostReviewStatus(error.response.status));
        }
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionsType.GET_REVIEWS:
      return Object.assign({}, state, {
        reviewsList: action.payload,
      });
    case ActionsType.POST_REVIEW:
      return Object.assign({}, state, {
        reviewsList: action.payload,
      });
    case ActionsType.GET_POST_REVIEW_STATUS:
      return Object.assign({}, state, {
        postReviewStatus: action.payload,
      });
  }
  return state;
};

export {reducer, ActionCreator, Operation};
