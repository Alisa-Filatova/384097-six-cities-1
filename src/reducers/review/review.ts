const initialState = {
  reviewsList: [],
  saveReviewStatus: null,
};

interface ActionType {
  type: ReviewAction,
  payload: any,
}

enum ReviewAction {
  GET_REVIEWS = 'GET_REVIEWS',
  ADD_REVIEW = 'ADD_REVIEW',
  SET_SAVE_REVIEW_STATUS = 'SET_SAVE_REVIEW_STATUS',
}

const ActionCreator = {
  getReviews: (reviews) => ({
    type: ReviewAction.GET_REVIEWS,
    payload: reviews,
  }),

  addReview: (review) => ({
    type: ReviewAction.ADD_REVIEW,
    payload: review,
  }),

  setSaveReviewStatus: (saveReviewStatus) => ({
    type: ReviewAction.SET_SAVE_REVIEW_STATUS,
    payload: saveReviewStatus,
  }),
};

const Operation = {
  getReviewsList: (id: number) => (dispatch, getState, api) => {
    return api.get(`/comments/${id}`)
      .then((response) => {
        dispatch(ActionCreator.getReviews(response.data));
      })
      .catch(() => {});
  },

  saveReview: (id: number, review) => (dispatch, getState, api) => {
    return api.post(`/comments/${id}`, review)
      .then((response) => {
        dispatch(ActionCreator.addReview(response.data));
        dispatch(ActionCreator.setSaveReviewStatus(response.status));
      })
      .catch((error) => {
        if (error) {
          dispatch(ActionCreator.setSaveReviewStatus(error.response.status));
        }
      });
  },
};

const reducer = (state = initialState, action: ActionType) => {
  switch (action.type) {
    case ReviewAction.GET_REVIEWS:
      return {
        ...state,
        reviewsList: action.payload,
      };

    case ReviewAction.ADD_REVIEW:
      return {
        ...state,
        reviewsList: action.payload,
      };

    case ReviewAction.SET_SAVE_REVIEW_STATUS:
      return {
        ...state,
        saveReviewStatus: action.payload,
      };

    default:
      return state;
  }
};

export {reducer, ActionCreator, Operation, ReviewAction, initialState};
