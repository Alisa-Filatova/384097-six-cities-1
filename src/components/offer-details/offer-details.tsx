import * as React from 'react';
import {connect} from 'react-redux';
import {Operation} from '../../reducers/review/review';
import {Operation as DataOperation} from '../../reducers/data/data';
import {sortReviewsByDate, getSaveReviewStatus} from '../../reducers/review/selectors';
import {getOfferById, getCloserOffers} from '../../reducers/data/selectors';
import {getAuthorizationStatus} from '../../reducers/user/selectors';
import ReviewsList from '../reviews-list/reviews-list';
import Map from '../map/map';
import OffersList from '../offers-list/offers-list';
import ReviewForm from '../review-form/review-form';
import FavoriteButton from '../favorite-button/favorite-button';
import RatingStars from '../rating-stars/rating-stars';
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import PlaceType from '../../types/enums/place-type';
import {MAX_CLOSER_OFFERS, MAX_PLACE_IMG, ROUTES} from '../../constants/constants';
import {Offer} from '../../types/offer';
import {Review} from '../../types/user';

const CloserOffersList = withActiveItem(OffersList);

interface Props {
  id: number;
  offer: Offer;
  nearOffers: Offer[];
  getReviews: () => void;
  reviews: Review[];
  isAuthenticated: boolean;
  saveReviewStatus: number;
  onFavoriteClick: (Offer) => void;
  history?: any[];
}

class OfferDetails extends React.PureComponent<Props> {

  componentDidMount() {
    this.props.getReviews();
  }

  componentDidUpdate(prevProps) {
    if (this.props.id !== prevProps.id) {
      this.props.getReviews();
    }
  }

  private handleFavoriteClick = () => {
    const {offer, onFavoriteClick, isAuthenticated, history} = this.props;

    if (isAuthenticated) {
      onFavoriteClick(offer);
    } else {
      history.push(ROUTES.LOGIN);
    }
  };

  render() {
    const {
      offer,
      nearOffers,
      reviews,
      isAuthenticated,
      onFavoriteClick,
      history,
      saveReviewStatus,
    } = this.props;

    if (!offer) {
      return null;
    }

    return (
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {offer.images.slice(0, MAX_PLACE_IMG).map((image, idx) =>
                <div className="property__image-wrapper" key={image + idx}>
                  <img className="property__image" src={image} alt="Photo studio" />
                </div>
              )}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {offer.isPremium &&
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              }
              <div className="property__name-wrapper">
                <h1 className="property__name">{offer.title}</h1>
                <FavoriteButton
                  isActive={offer.isFavorite}
                  onClick={this.handleFavoriteClick}
                  large
                />
              </div>
              <RatingStars
                rating={offer.rating}
                showValue
                prefix="property"
              />
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {PlaceType[offer.type.toUpperCase()]}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {offer.bedrooms} {offer.bedrooms === 1 ? 'Bedroom' : 'Bedrooms'}
                </li>
                <li className="property__feature property__feature--adults">
                  Max {offer.maxAdults} {offer.maxAdults === 1 ? 'adult' : 'adults'}
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {offer.goods.map((good) =>
                    <li key={good} className="property__inside-item">
                      {good}
                    </li>
                  )}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div
                    className={`property__avatar-wrapper ${offer.host.isPro ? 'property__avatar-wrapper--pro' : ''} 
                    user__avatar-wrapper`}
                  >
                    <img
                      className="property__avatar user__avatar"
                      src={`/${offer.host.avatarUrl}`}
                      width="74"
                      height="74"
                      alt="Host avatar"
                    />
                  </div>
                  <span className="property__user-name">{offer.host.name}</span>
                  {offer.host.isPro && <span className="property__user-status">Pro</span>}
                </div>
                <div className="property__description">
                  <p className="property__text">{offer.description}</p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <ReviewsList reviews={reviews} />
                {isAuthenticated && (
                  <ReviewForm
                    offerId={offer.id}
                    saveReviewStatus={saveReviewStatus}
                  />
                )}
              </section>
            </div>
          </div>
          <Map
            className="property__map map"
            key={offer.city.name}
            currentCity={offer.city}
            activeOfferId={offer.id}
            cityOffers={nearOffers.concat(offer)}
          />
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <CloserOffersList
              className="near-places__list places__list"
              rentalOffers={nearOffers}
              history={history}
              onFavoriteClick={onFavoriteClick}
              isAuthenticated={isAuthenticated}
            />
          </section>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const {id} = ownProps.match.params;

  return {
    ...ownProps,
    id,
    offer: getOfferById(state, id),
    nearOffers: getCloserOffers(state, id).slice(0, MAX_CLOSER_OFFERS),
    reviews: sortReviewsByDate(state),
    isAuthenticated: getAuthorizationStatus(state),
    saveReviewStatus: getSaveReviewStatus(state),
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  getReviews: () => {
    dispatch(Operation.getReviewsList(ownProps.match.params.id));
  },
  onFavoriteClick: (offer) => {
    dispatch(DataOperation.toggleFavorite(offer));
  },
});

export {OfferDetails};
export default connect(mapStateToProps, mapDispatchToProps)(OfferDetails);
