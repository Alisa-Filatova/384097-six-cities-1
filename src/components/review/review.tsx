import * as React from 'react';
import RatingStars from '../rating-stars/rating-stars';
import {Review as ReviewType} from '../../types/user';

interface Props {
  review: ReviewType;
}

const Review: React.FunctionComponent<Props> = ({review}) => {
  const reviewDate = new Date(review.date).toLocaleDateString(`en-US`, {
    month: `long`,
    year: `numeric`,
  });

  return (
    <li className="reviews__item">
      <div className="reviews__user user">
        <div className="reviews__avatar-wrapper user__avatar-wrapper">
          <img
            className="reviews__avatar user__avatar"
            src={review.user.avatarUrl}
            width="54"
            height="54"
            alt="Reviews avatar"
          />
        </div>
        <span className="reviews__user-name">{review.user.name}</span>
      </div>
      <div className="reviews__info">
        <RatingStars
          rating={review.rating}
          prefix="reviews"
        />
        <p className="reviews__text">{review.comment}</p>
        <time className="reviews__time" dateTime={review.date}>
          {reviewDate}
        </time>
      </div>
    </li>
  );
};

export default Review;
