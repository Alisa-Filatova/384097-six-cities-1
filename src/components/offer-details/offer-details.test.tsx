import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {OfferDetails} from './offer-details';
import {offersMock} from '../../mocks/offers';
import {offerMock} from '../../mocks/offer';
import {Offer} from '../../types/offer';
import {reviewMock} from '../../mocks/review';
import {Review} from '../../types/user';

configure({adapter: new Adapter()});

describe(`OfferDetails`, () => {
  it(`renders correctly`, () => {
    const page = shallow(
        <OfferDetails
          nearOffers={offersMock as Offer[]}
          reviews={[reviewMock as Review]}
          offer={offerMock as Offer}
          onFavoriteClick={jest.fn()}
          isAuthenticated={true}
          getReviews={jest.fn()}
          saveReviewStatus={0}
          id={4}
        />
    );

    expect(shallowToJson(page)).toMatchSnapshot();
  });
});
