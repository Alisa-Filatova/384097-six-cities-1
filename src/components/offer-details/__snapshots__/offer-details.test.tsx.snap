// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferDetails renders correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
          key="img/1.png0"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/1.png"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="img/2.png1"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/2.png"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Beautiful & luxurious studio at great location
          </h1>
          <FavoriteButton
            isActive={false}
            large={true}
            onClick={[Function]}
          />
        </div>
        <RatingStars
          prefix="property"
          rating={4.8}
          showValue={true}
        />
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             
            Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            4
             
            adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            120
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
              key="Heating"
            >
              Heating
            </li>
            <li
              className="property__inside-item"
              key="Kitchen"
            >
              Kitchen
            </li>
            <li
              className="property__inside-item"
              key="Cable TV"
            >
              Cable TV
            </li>
            <li
              className="property__inside-item"
              key="Washing machine"
            >
              Washing machine
            </li>
            <li
              className="property__inside-item"
              key="Coffee machine"
            >
              Coffee machine
            </li>
            <li
              className="property__inside-item"
              key="Dishwasher"
            >
              Dishwasher
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro 
                    user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/img/1.png"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <ReviewsList
            reviews={
              Array [
                Object {
                  "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "date": "2019-05-08T14:13:56.569Z",
                  "id": 1,
                  "rating": 4,
                  "user": Object {
                    "avatarUrl": "img/1.png",
                    "id": 4,
                    "isPro": false,
                    "name": "Max",
                  },
                },
              ]
            }
          />
          <ConnectFunction
            offerId={1}
            saveReviewStatus={0}
          />
        </section>
      </div>
    </div>
    <Map
      activeOfferId={1}
      cityOffers={
        Array [
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Amsterdam",
            },
            "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
            "goods": Array [
              "Heating",
              "Kitchen",
              "Cable TV",
              "Washing machine",
              "Coffee machine",
              "Dishwasher",
            ],
            "host": Object {
              "avatarUrl": "img/1.png",
              "id": 3,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "img/1.png",
              "img/2.png",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.35514938496378,
              "longitude": 4.673877537499948,
              "zoom": 8,
            },
            "maxAdults": 4,
            "previewImage": "img/1.png",
            "price": 120,
            "rating": 4.8,
            "title": "Beautiful & luxurious studio at great location",
            "type": "apartment",
          },
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Paris",
            },
            "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
            "goods": Array [
              "Heating",
              "Kitchen",
              "Cable TV",
              "Washing machine",
              "Coffee machine",
              "Dishwasher",
            ],
            "host": Object {
              "avatarUrl": "img/1.png",
              "id": 4,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 2,
            "images": Array [
              "img/1.png",
              "img/2.png",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.35514938496378,
              "longitude": 4.673877537499948,
              "zoom": 8,
            },
            "maxAdults": 4,
            "previewImage": "img/1.png",
            "price": 120,
            "rating": 4.8,
            "title": "Beautiful & luxurious studio at great location",
            "type": "apartment",
          },
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Amsterdam",
            },
            "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
            "goods": Array [
              "Heating",
              "Kitchen",
              "Cable TV",
              "Washing machine",
              "Coffee machine",
              "Dishwasher",
            ],
            "host": Object {
              "avatarUrl": "img/1.png",
              "id": 3,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "img/1.png",
              "img/2.png",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.35514938496378,
              "longitude": 4.673877537499948,
              "zoom": 8,
            },
            "maxAdults": 4,
            "previewImage": "img/1.png",
            "price": 120,
            "rating": 4.8,
            "title": "Beautiful & luxurious studio at great location",
            "type": "apartment",
          },
        ]
      }
      className="property__map map"
      currentCity={
        Object {
          "location": Object {
            "latitude": 52.370216,
            "longitude": 4.895168,
            "zoom": 10,
          },
          "name": "Amsterdam",
        }
      }
      key="Amsterdam"
    />
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <WithActiveItem
        className="near-places__list places__list"
        isAuthenticated={true}
        onFavoriteClick={[MockFunction]}
        rentalOffers={
          Array [
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/1.png",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "maxAdults": 4,
              "previewImage": "img/1.png",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Paris",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/1.png",
                "id": 4,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "img/1.png",
                "img/2.png",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.35514938496378,
                "longitude": 4.673877537499948,
                "zoom": 8,
              },
              "maxAdults": 4,
              "previewImage": "img/1.png",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "apartment",
            },
          ]
        }
      />
    </section>
  </div>
</main>
`;
